src/commands/tag.js

```diff
@@ -0,0 +1,114 @@
+import { writeObject } from './writeObject'
+import { readObject } from './readObject'
+import { writeRef } from './writeRef'
+import { config } from './config'
+import { GitRefManager } from '../managers/GitRefManager'
+import { E, GitError } from '../models/GitError.js'
+import { FileSystem } from '../models/FileSystem.js'
+import { join } from '../utils/join.js'
+import { cores } from '../utils/plugins.js'
+import { GitAnnotatedTag } from '../models/GitAnnotatedTag'
+
+/**
+ * Create a lightweight or annotated tag.
+ *
+ * @link https://isomorphic-git.github.io/docs/tag.html
+ */
+export async function tag ({
+  core = 'default',
+  dir,
+  gitdir = join(dir, '.git'),
+  fs: _fs = cores.get(core).get('fs'),
+  name,
```

Comment:
I think it'd better rename too.

FWIW `branch` and `deleteBranch` seem to use `ref` instead of `branch` 😭

Reply:
I agree with `tag` here. (Though, if you want to align with the git docs, it would be tagname).

I think a case could be name that the branch/deleteBranch commands should use the argument name `branch` (or `branchname`) instead of `ref`.

Reply:
Yeah, we're building up some inconsistency lol. Why is `addRemote` not called `remote` etc? I'll make an issue to try to figure out what's the best way forward.

Reply:
see #631